### PR TITLE
GLT-2817: allow editing sample classes

### PR DESCRIPTION
**Pull request by djcooke**
Will update user manual after the documentation makeover PR is merged

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleIdentity.java [hunk 1]
```diff
@@ -11,6 +12,8 @@
   
   public static final String CATEGORY_NAME = "Identity";
 
+  public static final List<String> SUBCATEGORIES = Collections.unmodifiableList(Collections.emptyList());
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleIdentity.java]: The empty list is implicitly unmodifable.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java [hunk 2]
```diff
@@ -4,68 +4,65 @@
...
+  public void setArchived(boolean archived);
 
-  boolean isArchived();
+  public boolean isDirectCreationAllowed();
 
-  void setArchived(boolean archived);
+  public void setDirectCreationAllowed(boolean directCreationAllowed);
 
-  boolean isDirectCreationAllowed();
+  public boolean getDNAseTreatable();
```
Comment (avarsava) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java]: It looks like everything else is using 'is__' for boolean getters, is there a reason this method (and others named the same thing) still use 'get'?

File: miso-web/src/main/webapp/scripts/form_sampleclass.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          maxLength: 255
+        },
+        {
+          title: 'Category',
+          data: 'sampleCategory',
+          type: 'dropdown',
+          required: true,
+          getSource: function() {
+            return Constants.sampleCategories;
+          },
```
Comment (apmasell) [inline miso-web/src/main/webapp/scripts/form_sampleclass.js]: Is this not going to be bad for the DB if you change it?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java (hunk 2 above)
Comment (djcooke) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java]: It hadn't annoyed me enough to rename it? We don't consistently use one or the other throughout MISO

Comment (avarsava) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java]: Alright! Just noticed getArchived() became isArchived() and this one didn't :+1:

File: miso-web/src/main/webapp/scripts/form_sampleclass.js (hunk 3 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/form_sampleclass.js]: There's validation in the service layer that prevents changing it if the class is used by any samples, or if relationships would become invalid

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java (hunk 2 above)
Comment (djcooke) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SampleClass.java]: `getArchived` had annoyed me enough 😆

**Pull request merged**